autowire/shared/src/test/scala/autowire/UpickleTests.scala

```diff
@@ -1,288 +1,311 @@
-                                                              ^
-        """,
+          |      test - compileError("Client[Api].add(1, 2, 3).toString.call()").check(
+          |                                                                 ^
+        """.stripMargin,
         "You can't call the .call() method",
         "add(1, 2, 3).toString()"
       )
-
-      * - compileError("Client[Api].fail1().call()").check(
+      test - compileError("Client[Api].fail1().call()").check(
         """
-      * - compileError("Client[Api].fail1().call()").check(
-                                    ^
+          |      test - compileError("Client[Api].fail1().call()").check(
+          |                                       ^
         """.stripMargin,
         "value fail1 is not a member of autowire.ClientProxy"
       )
     }
-    test("runtimeFailures"){
-      test("noSuchRoute"){
+
+    test("runtimeFailures") - {
+      test("noSuchRoute") {
         val badRequest = Core.Request[String](Seq("omg", "wtf", "bbq"), Map.empty)
         assert(!Server.routes.isDefinedAt(badRequest))
         intercept[MatchError] {
           Server.routes(badRequest)
         }
       }
-      test("inputError"){
-        def check(input: Map[String, String])(expectedError: PartialFunction[Throwable, Unit]) = {
+
+      test("inputError") - {
+        def check(input: Map[String, String])(expectedError: PartialFunction[Throwable, Unit]): Unit = {
           val badRequest = Core.Request(
             Seq("autowire", "Api", "multiply"),
             input
           )
           assert(Server.routes.isDefinedAt(badRequest))
-          val error = intercept[Error] { Server.routes(badRequest) }
+          val error = intercept[Error] {
+            Server.routes(badRequest)
+          }
           assert(expectedError.isDefinedAt(error))
         }
 
-        test("keysMissing") {
-          * - check(Map.empty){
+        test("keysMissing") - {
+          test - check(Map.empty) {
             case Error.InvalidInput(
-              Error.Param.Missing("x"),
-              Error.Param.Missing("ys")
+            Error.Param.Missing("x"),
+            Error.Param.Missing("ys")
             ) =>
           }
-          * - check(Map("x" -> "123")){
+          test - check(Map("x" -> "123")) {
             case Error.InvalidInput(Error.Param.Missing("ys")) =>
           }
-          * - check(Map("ys" -> "[123]")){
+          test - check(Map("ys" -> "[123]")) {
             case Error.InvalidInput(Error.Param.Missing("x")) =>
           }
-
         }
+
         test("keysInvalid") - {
-          * - check(Map("x" -> "[]", "ys" -> "234")) {
+          test - check(Map("x" -> "[]", "ys" -> "234")) {
             case Error.InvalidInput(
-              Error.Param.Invalid("x", InvalidData(Arr(Seq()), _)),
-              Error.Param.Invalid("ys", InvalidData(Num(234), _))
+            Error.Param.Invalid("x", AbortException("expected number got sequence", _, _, _, _)),
```

Comment:
Adding the error messages helps understand the problems.

Reply:
I hadn't add Exception strings in tests, this is better this way